pdf/pdfannotation.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2016 Caliste Damien.
+ * Contact: Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; version 2 only.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#ifndef PDFANNOTATION_H
+#define PDFANNOTATION_H
+
+#include <poppler-qt5.h>
+
+#include "pdfdocument.h"
+#include "pdfselection.h"
+
+class PDFTextAnnotation;
+class PDFHighlightAnnotation;
+
+class PDFAnnotation : public QObject
+{
+    Q_OBJECT
+
+    Q_ENUMS(SubType)
+
+    Q_PROPERTY(PDFDocument* document READ document NOTIFY attached)
+    Q_PROPERTY(int page READ page NOTIFY attached)
+    Q_PROPERTY(QRectF boundary READ boundary NOTIFY attached)
+
+    Q_PROPERTY(QString author READ author WRITE setAuthor NOTIFY authorChanged)
+    Q_PROPERTY(QString contents READ contents WRITE setContents NOTIFY contentsChanged)
+    Q_PROPERTY(QDateTime creationDate READ creationDate CONSTANT)
+    Q_PROPERTY(QDateTime modificationDate READ modificationDate NOTIFY modificationDateChanged)
+    Q_PROPERTY(QColor color READ color WRITE setColor NOTIFY colorChanged)
+    Q_PROPERTY(SubType type READ type CONSTANT)
+
```

Comment:
You're right. Initially, I've bound as much of Poppler-Qt Annotation object as possible from the C++ side, because it was 1:1 code. Then, the UI part is finally not using all capabilities exposed from C++. I didn't have envy to just erase all this code, even if not use, in case it will be later.

From code development point of view, it's dead code, indeed. I will remove it if you think it's better like that (*i.e.* less maintenance burden).